### DISCUSSION TITLE
What are the current best practices for generating automatic API documentation from Blitz?

### DISCUSSION THREAD

**Discussion by mkondel** [blitz-js/blitz#3182] [Q&A] [answered]
If you have any suggestions on how to generate an openapi.json file from a Blitz app, please post them in here. Are there any existing solutions to this? I found very few issues/discussions here on this topic. Some are very old at this point, and none of them go very far.

I am hoping to find a way to make this happen automatically, based only on the code for the routes. Maybe leveraging typescript somehow. On the request side we have Zod, so it may be possible to generate something based on those schemas. On the response side there is nothing, but there is that `resolver()`. If it knew a return type, then maybe it could also be used to generate something.

**Comment by beerose** [comment, marked as answer]
I don't think that there's any update on that, but it's possibly something we can tackle in the future with the toolkit (https://github.com/blitz-js/blitz/discussions/3075) as a new feature.

**Reply by mkondel** [reply]
Thank you for confirming that. I am guessing that the easiest way right now is to use JSDoc comments to build the openapi.json file. The "manual" way.
